File: app/src/main/java/com/viginfotech/chennaitimes/data/NewsContract.java
package com.viginfotech.chennaitimes.data;

import android.content.ContentUris;
import android.net.Uri;
import android.provider.BaseColumns;

/**
 * Created by anand on 9/25/15.
 */
public class NewsContract {

    public static final String CONTENT_AUTHORITY="com.chennai.times.app";
    public static final Uri BASE_CONTENT_URI= Uri.parse("content://"+CONTENT_AUTHORITY);

    public static final String PATH_DB="newsapp";

    public static final class NewsEntry implements BaseColumns {
        public static final Uri CONTENT_URI =
                BASE_CONTENT_URI.buildUpon().appendPath(PATH_DB).build();

        public static final String CONTENT_TYPE =
                "vnd.android.cursor.dir/" + CONTENT_AUTHORITY + "/" + PATH_DB;
        public static final String CONTENT_ITEM_TYPE =
                "vnd.android.cursor.item/" + CONTENT_AUTHORITY + "/" + PATH_DB;

        public static final String TABLE_NAME="feed";

        public static final String COLUMN_FEED_TITLE="title";
        public static final String COLUMN_FEED_DETAILED_TITLE="detailedTitle";
        public static final String COLUMN_FEED_SUMMARY="summary";
        public static final String COLUMN_FEED_PUBDATE="pubDate";
        public static final String COLUMN_FEED_LINK="guid";
        public static final String COLUMN_FEED_THUMBNAIL="thumbnail";
        public static final String COLUMN_FEED_IMAGE="image";
        public static final String COLUMN_FEED_DETAILED_NEWS="detailNews";
        public static final String COLUMN_FEED_CATEGORY="category";
        public static final String COLUMN_FEED_SOURCE="source";
        public static final String COLUMN_READ_STATE="readState";

        public static Uri buildFeedUri(long id) {
            return ContentUris.withAppendedId(CONTENT_URI, id);
        }

    }
}

File: app/src/main/java/com/viginfotech/chennaitimes/data/NewsContract.kt
package com.viginfotech.chennaitimes.data

import android.content.ContentUris
import android.net.Uri
import android.provider.BaseColumns
import com.viginfotech.chennaitimes.BuildConfig

/**
 * Created by anand on 9/25/15.
 */
object NewsContract {

    val CONTENT_AUTHORITY = if(BuildConfig.DEBUG)
        "com.viginfotech.chennaitimes.app.debug"
            else "com.viginfotech.chennaitimes.app"
    val BASE_CONTENT_URI = Uri.parse("content://$CONTENT_AUTHORITY")

    val PATH_DB = "newsapp"

    class NewsEntry : BaseColumns {
        companion object {
            val _ID = "_id"
            val CONTENT_URI = BASE_CONTENT_URI.buildUpon().appendPath(PATH_DB).build()

            val CONTENT_TYPE =
                    "vnd.android.cursor.dir/$CONTENT_AUTHORITY/$PATH_DB"
            val CONTENT_ITEM_TYPE =
                    "vnd.android.cursor.item/$CONTENT_AUTHORITY/$PATH_DB"

            val TABLE_NAME = "feed"

            val COLUMN_FEED_TITLE = "title"
            val COLUMN_FEED_DETAILED_TITLE = "detailedTitle"
            val COLUMN_FEED_SUMMARY = "summary"
            val COLUMN_FEED_PUBDATE = "pubDate"
            val COLUMN_FEED_LINK = "guid"
            val COLUMN_FEED_THUMBNAIL = "thumbnail"
            val COLUMN_FEED_IMAGE = "image"
            val COLUMN_FEED_DETAILED_NEWS = "detailNews"
            val COLUMN_FEED_CATEGORY = "category"
            val COLUMN_FEED_SOURCE = "source"
            val COLUMN_READ_STATE = "readState"

            fun buildFeedUri(id: Long): Uri {
                return ContentUris.withAppendedId(CONTENT_URI, id)
            }
        }

    }
}
